### DISCUSSION TITLE
Thank you

### DISCUSSION THREAD

**Discussion by DoughMucker** [supersaiyanmode/PyWebOSTV#78] [General]
I just wanted to pass on some thanks for this. I'm using it to control my OLED83C2 which is connected to a CoreELEC box (Kodi). I configured it so remote button presses start python scripts which change inputs, control power, etc.

Doug

**Comment by terrorindeed** [comment]
amazing! 
I am on a simliar path to control my oppo 203 via the Vero 4k+. Could you answer some questions for me? I am a novice python coder. I will say I've learned a lot from chatgpt, in 25 days. lol

**Comment by DoughMucker** [comment]
A better place to ask your questions is on the Kodi or Vero forums (as it's not webostv related.) Better yet, search the forums first as what you want is probably there already.

**Comment by terrorindeed** [comment]
As this is 100% pywebostv related, I thought someone would be kind and helpful. my apologies. 
I have my vero 4k+ triggering an http reg request to my opoo and playback works flawlessly. I was merely attempting to have someone better my exisiting scripts. 
Sadly, no one on the vero or kodi forums has any interest in this project. 

If I can start again, 
When I initilize source_control.set_source(sources[2]), the tv does switch hdmi inputs correctly, but the time delay is roughly 15 seconds. On the otherhand, my script to initiate the http get request to the oppo is essentially instanteous. Without writing a delay in my oppo playback script, is there anything i can do to increase the speediness of the webostv command?

**Reply by DoughMucker** [reply]
maybe use timer, timeit, time.process_timer() or some other method to track which line of code is taking too much time.
